inc/core/styles/frontend.php

```diff
@@ -1140,6 +1141,98 @@ private function setup_single_post_style() {
+				Dynamic_Selector::META_KEY => Config::MODS_POST_COMMENTS_BACKGROUND_COLOR,
+			],
+			'--color'   => [
+				Dynamic_Selector::META_KEY => Config::MODS_POST_COMMENTS_TEXT_COLOR,
+			],
+		];
+
+		$this->_subscribers[] = [
+			'selectors' => '.nv-is-boxed.nv-comments-wrap',
+			'rules'     => $boxed_comments_rules,
+		];
+
+		$boxed_comment_form_rules = [
+			'--padding' => [
+				Dynamic_Selector::META_KEY           => Config::MODS_POST_COMMENTS_FORM_PADDING,
+				Dynamic_Selector::META_IS_RESPONSIVE => true,
+				Dynamic_Selector::META_DEFAULT       => $this->padding_default(),
+				'directional-prop'                   => Config::CSS_PROP_PADDING,
+			],
+			'--bgColor' => [
+				Dynamic_Selector::META_KEY => Config::MODS_POST_COMMENTS_FORM_BACKGROUND_COLOR,
+			],
+			'--color'   => [
+				Dynamic_Selector::META_KEY => Config::MODS_POST_COMMENTS_FORM_TEXT_COLOR,
+			],
+		];
+
+		$this->_subscribers[] = [
+			'selectors' => '.nv-is-boxed.comment-respond',
+			'rules'     => $boxed_comment_form_rules,
+		];
+
+		$spacing_rules = [
+			'--spacing' => [
+				Dynamic_Selector::META_KEY           => Config::MODS_SINGLE_POST_ELEMENTS_SPACING,
+				Dynamic_Selector::META_IS_RESPONSIVE => true,
+				Dynamic_Selector::META_SUFFIX        => 'px',
+			],
+		];
+
+		$this->_subscribers[] = [
+			'selectors' => '.nv-single-post-wrap',
+			'rules'     => $spacing_rules,
+		];
+	}
+
+	/**
+	 * Add css for post/page header.
+	 */
+	private function setup_header_style() {
+
+		$context = $this->get_header_context();
+		if ( empty( $context ) ) {
+			return;
+		}
+
+		$dataset = [
+			'height'           => Config::MODS_POST_COVER_HEIGHT,
+			'padding'          => Config::MODS_POST_COVER_PADDING,
+			'title_alignment'  => Config::MODS_POST_COVER_TITLE_ALIGNMENT,
+			'title_position'   => Config::MODS_POST_COVER_TITLE_POSITION,
+			'text_color'       => Config::MODS_POST_COVER_TEXT_COLOR,
+			'background_color' => Config::MODS_POST_COVER_BACKGROUND_COLOR,
+			'blend_mode'       => Config::MODS_POST_COVER_BLEND_MODE,
+			'opacity'          => Config::MODS_POST_COVER_OVERLAY_OPACITY,
+			'boxed_padding'    => Config::MODS_POST_COVER_BOXED_TITLE_PADDING,
+			'boxed_background' => Config::MODS_POST_COVER_BOXED_TITLE_BACKGROUND,
+		];
+		if ( $context === 'page' ) {
+			$dataset = [
+				'height'           => Config::MODS_PAGE_COVER_HEIGHT,
+				'padding'          => Config::MODS_PAGE_COVER_PADDING,
+				'title_alignment'  => Config::MODS_PAGE_COVER_TITLE_ALIGNMENT,
+				'title_position'   => Config::MODS_PAGE_COVER_TITLE_POSITION,
+				'text_color'       => Config::MODS_PAGE_COVER_TEXT_COLOR,
+				'background_color' => Config::MODS_PAGE_COVER_BACKGROUND_COLOR,
+				'blend_mode'       => Config::MODS_PAGE_COVER_BLEND_MODE,
+				'opacity'          => Config::MODS_PAGE_COVER_OVERLAY_OPACITY,
+				'boxed_padding'    => Config::MODS_PAGE_COVER_BOXED_TITLE_PADDING,
+				'boxed_background' => Config::MODS_PAGE_COVER_BOXED_TITLE_BACKGROUND,
+			];
+		}
```

Comment:
I thing you could use `constant("Config::MODS_" . strtoupper($context) . "_$meta" )` ore something similar and just get the constants dynamically.  What do you think @cristian-ungureanu ? @abaicus What do you think, would over-complicate things?

You could have this logic in a function that you pass the meta and context and will return the config value

Reply:
I think we can do it with the constant function, provided that the $context variable value is predictable. Also, let's make sure that in the bail case `get_header_context` returns an empty string.